Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -1511,96 +863,28 @@ void LoadEventNexus::loadEvents(API::Progress *const prog,
   longest_tof = 0.;
 
-  // Make the thread pool
-  ThreadScheduler *scheduler = new ThreadSchedulerMutexes();
-  ThreadPool pool(scheduler);
-  auto diskIOMutex = boost::make_shared<std::mutex>();
-  size_t bank0 = 0;
-  size_t bankn = bankNames.size();
-
-  if (chunk !=
-      EMPTY_INT()) // We are loading part - work out the bank number range
-  {
-    eventsPerChunk = total_events / totalChunks;
-    // Sort banks by size
-    size_t tmp;
-    string stmp;
-    for (size_t i = 0; i < bankn; i++)
-      for (size_t j = 0; j < bankn - 1; j++)
-        if (bankNumEvents[j] < bankNumEvents[j + 1]) {
-          tmp = bankNumEvents[j];
-          bankNumEvents[j] = bankNumEvents[j + 1];
-          bankNumEvents[j + 1] = tmp;
-          stmp = bankNames[j];
-          bankNames[j] = bankNames[j + 1];
-          bankNames[j + 1] = stmp;
-        }
-    int bigBanks = 0;
-    for (size_t i = 0; i < bankn; i++)
-      if (bankNumEvents[i] > eventsPerChunk)
-        bigBanks++;
-    // Each chunk is part of bank or multiple whole banks
-    // 0.5 for last chunk of a bank with multiple chunks
-    // 0.1 for multiple whole banks not completely filled
-    eventsPerChunk +=
-        static_cast<size_t>((static_cast<double>(bigBanks) /
-                                 static_cast<double>(totalChunks) * 0.5 +
-                             0.05) *
-                            static_cast<double>(eventsPerChunk));
-    double partialChunk = 0.;
-    firstChunkForBank = 1;
-    for (int chunki = 1; chunki <= chunk; chunki++) {
-      if (partialChunk > 1.) {
-        partialChunk = 0.;
-        firstChunkForBank = chunki;
-        bank0 = bankn;
-      }
-      if (bankNumEvents[bank0] > 1) {
-        partialChunk += static_cast<double>(eventsPerChunk) /
-                        static_cast<double>(bankNumEvents[bank0]);
-      }
-      if (chunki < totalChunks)
-        bankn = bank0 + 1;
-      else
-        bankn = bankNames.size();
-      if (chunki == firstChunkForBank && partialChunk > 1.0)
-        bankn += static_cast<size_t>(partialChunk) - 1;
-      if (bankn > bankNames.size())
-        bankn = bankNames.size();
-    }
-    for (size_t i = bank0; i < bankn; i++) {
-      size_t start_event = (chunk - firstChunkForBank) * eventsPerChunk;
-      size_t stop_event = bankNumEvents[i];
-      // Don't change stop_event for the final chunk
-      if (start_event + eventsPerChunk < stop_event)
-        stop_event = start_event + eventsPerChunk;
-      bankNumEvents[i] = stop_event - start_event;
+  bool loaded{false};
+  if (canUseParallelLoader(haveWeights, oldNeXusFileNames, classType)) {
+    auto ws = m_ws->getSingleHeldWorkspace();
+    m_file->close();
+    try {
+      ParallelEventLoader::load(*ws, m_filename, m_top_entry_name, bankNames);
+      loaded = true;
+      shortest_tof = 0.0;
+      longest_tof = 1e10;
+    } catch (const std::runtime_error &) {
     }
+    safeOpenFile(m_filename);
```

Comment:
Is this in the right place? We only need to open the file after the parallel loader?

Reply:
`LoadEventNeus` is opening the file via the Nexus API before reaching this piece of code. The parallel loader uses HDF5 directly and I didn't get it to work without first closing the file via the Nexus API, then let the parallel loader open and close it via HDF5, and in the end reopen it here such that `LoadEventNexus` can proceed.